tools/stats/monitor.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+import datetime
+import importlib.util
+import json
+import subprocess
+import sys
+import time
+from typing import Any, Dict, List
+
+
+def pip_install(package_name: str) -> None:
+    if importlib.util.find_spec(package_name) is None:
+        output = subprocess.run(
+            [sys.executable, "-m", "pip", "install", package_name],
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+        )
+        if output.returncode != 0:
+            print(output.stderr.decode("utf-8"))
+            print(output.stdout.decode("utf-8"))
+    # when running this script in the background, pip installs the packages to
+    # a location that is not in a path python uses to look for modules, so we
+    # manually add the location to the path
+    location = subprocess.run(
+        [sys.executable, "-m", "pip", "show", package_name],
+        stderr=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+    )
+    for s in location.stdout.decode("utf-8").splitlines():
+        if "Location:" in s:
+            sys.path.append(s.split(" ")[1])
+
+
+def main() -> None:
+    pip_install("psutil")
```

Comment:
Because `psutil` and `pynvml` are used across the board (linux, mac, rocm, win) from what I see here. Do you think it make more sense to reuse [pip_install](https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/common_utils.sh#L52-L57) instead of hand rolling one yourselves here? Less duplication is better I guess

Reply:
edited to just pip installing it in the github actions step instead